### PR TITLE
refactor[core]: support nx 16

### PR DESCRIPTION
**Pull request by gmfun**

### CODE REVIEW

**Comment by NiklasPor** [comment]
Looks good!

**Review by RodrigoTerek — APPROVED** [review]

File: package.json
Comment (RodrigoTerek) [inline package.json:1]: The version shouldn't  be updated as well?

**Comment by rathpc** [comment]
@gmfun thanks for opening this PR I was just about to!

For the maintainers, any idea of when this can get merged to create a new release?

File: package.json
Comment (rathpc) [reply package.json:1]: it should definitely be a major bump but I am not sure the process taken for releases here

**Review by rathpc — APPROVED** [review]
nice!

**Comment by RodrigoTerek** [comment]
@NiklasPor do you have an idea of when will this be merged? I am also depending on this.

**Comment by NiklasPor** [comment]
In about 2 hours, I'm currently preparing a release of another package and then continue with taking care of this one.

File: package.json
Comment (NiklasPor) [reply package.json:1]: I'll take care of version bumping :) 👍

File: lib/create-custom-runner.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { RemoteCache } from "@nrwl/workspace/src/tasks-runner/default-tasks-runner";
-import defaultTasksRunner from "@nrwl/workspace/tasks-runners/default";
```
[the remark this answers is not in the document]
Comment (NiklasPor) [reply lib/create-custom-runner.ts:2]: As the peer dependency support starts with 16 and we'll have to create a major bump anyway, I'd say we just keep nx here for the sake of simplicity.

**Comment by NiklasPor** [comment]
Looks good, thanks for the contribution @gmfun 👏 👏 I'll now check that nothing breaks in the other package and then create a few releases :)

**Comment by NiklasPor** [comment]
Version `4.0.0` for `nx-remotecache-custom` is published (I always need to setup a new azure container and stuff for testing the integration 😆 ). `nx-remtoecache-azure@4.0.0` is also on it's way.

**Comment by NiklasPor** [comment]
And thank you all for your patience, I'm currently in one job and also trying to build up a startup in the digital construction industry, so there's not so much time left besides that stuff 😵

**Comment by NiklasPor** [comment]
`nx-remtoecache-azure@4.0.0` and the `minio` (no one uses this one anyway) are also updated. If you depend on any other external package, which depends on this one, please notify the author 👍

**Pull request merged**